### ISSUE TITLE
Unable to access over 9 dozzle hosts/agents

### ISSUE THREAD

**Issue by beastarman74** [amir20/dozzle#3072]
**Describe the bug**
Unable to access dozzle hosts or agents after the first 9 are added. They cannot be accessed from the host/agent list or the drop down box. This appears to be the same in dozzle mobile and desktop view.

**To Reproduce**

1. run one or more full version of dozzle on docker hosts
2. run dozzle agent mode on over 10 Docker hosts including those running full dozzle
3. Open the dozzle Web page on one of the host's running full dozzle 
4. Attempt to switch to the 10th or 10th plus dozzle agent, it will not be accessible unless one of the first 9 is shutdown and the full dozzle instance restarted.

**Expected behavior**
All dozzle agent instances can be accessed from the dozzle Web page.

**Screenshots**
![Screenshot_20240708_002453](https://github.com/amir20/dozzle/assets/44511355/52f6458b-e6ce-49d7-80d8-103da89ea2b3)
![Screenshot_20240708_001149](https://github.com/amir20/dozzle/assets/44511355/640fc59a-82d7-477d-8747-8eea941008cf)
![Screenshot_20240708_000926](https://github.com/amir20/dozzle/assets/44511355/69ed39d0-bd16-4603-adfa-13cb259c9f05)

**Desktop (please complete the following information):**

- OS: Android 10 client, docker hosts all linux debian 11 (or raspberry pi variant) or ubuntu 22.04 LTS
- Docker version:
```
Client: Docker Engine - Community
 Version:           27.0.3
 API version:       1.46
 Go version:        go1.21.11
 Git commit:        7d4bcd8
 Built:             Sat Jun 29 00:02:44 2024
 OS/Arch:           linux/arm64
 Context:           default 
```
- Browser & version: Chrome Dev 128.0.6557.3

- Version: v8.0.1

**If applicable include logs with `--level debug` and browser logs**

**Comment by amir20** [comment]
> it will not be accessible unless one of the first 9 is shutdown

Do you mean it doesn't show up in the drop down or the UI doesn't work at all? 

Are you by any chance running HTTP1.1 which is limited. See https://stackoverflow.com/questions/16852690/sseeventsource-why-no-more-than-6-connections

I'll probably can look at this in the morning but a little more clarification is needed. 

Thanks for stress testing!

**Comment by beastarman74** [comment]
I believe it is a interface issue. 3 of the 12 hosts running dozzle agent are not visible in the list of agents shown in the main display or the host selection drop down menu. Other than that Dozzle and the listed agents appear fully functional from a log viewing and container action perspective.

I have 12 hosts running dozzle in agent mode with 3 of 
which running full dozzle as well.  I have 10 hosts in my homelab, 8 physical (Raspberry Pi, NanoPi and repurposed thin client) and 2 lxc containers running on proxmox on a repurposed micro business pc. I created additional test lxc containers to test further.

dockge shows the agents and full dozzle installed:
![Screenshot_20240708_111824](https://github.com/amir20/dozzle/assets/44511355/d512ecf2-495b-4032-9b8d-9c3702edc01c)

If I remove one of the listed agents and restart/recreate the full dozzle containers one of the dozzle agents will be shown in its place. I think it might use creation or start time to pick which but this could just be my perception.

Sorry my initial issue was not clear enough. It was my 4th attempt to create it, I lost it 3 times to chrome refreshing after a short period of inactivity so it likely wasnt my best effort.

Thanks for looking into this. 

Paul.

**Comment by amir20** [comment]
Can you check the logs when dozzle starts? Do you see any warnings?

Right now if agents fail at start up, they will not be retried. 

So I am wondering if some of the clients are not connecting.

**Comment by beastarman74** [comment]
I think I have led you on  a merry dance. There is something in the Full Dozzle log, I cannot believe I missed this.  It is the LXC containers. I have to use backup / restore method of creating the containers.  So I have a LXC template setup for this purpose. It has had Dozzle running in the past, in both modes (along with other containers, portainer agent, watchtower).  When I restore the backup to a new LXC Container I always ensure I tick the unique box. I believed this ensured all IDs where changed. 

It looks like Dozzle is remembering a host ID. But I cannot find where this is being pulled from stored.  

![image](https://github.com/amir20/dozzle/assets/44511355/8414f9d6-89c5-4d85-8169-d57f5f5ce2ab)

**Comment by amir20** [comment]
Ah there we go. This is coming from `docker system info`. If two instance have the same id then they would be treated as a duplicate. 

Is there a better id to use? According to Docker documentation, this is supposed to be a unique ID.

**Comment by beastarman74** [comment]
I have managed to sort the issue.  It needs the `engine-id` file that is stored in `/var/lib/docker ` removing from any machine with a duplicate ID.  Then restart docker or the host machine which will recreate it with a new value.

All the machines are now visible in Dozzle.
![image](https://github.com/amir20/dozzle/assets/44511355/a8b60b46-5ed8-4916-bcb7-7127b03e48f8)

I am sorry I have wasted your time on this one.  Not sure if it is worth renaming this issue to something similar to the below in case someone else in 50 years time manages to create the same issue for themselves! :
`WARNING [0001] duplicate host ID: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx. Endpoint:	xxx.xxx.xxx.xxx:7007 found, skipping`
